### DISCUSSION TITLE
What is best practice to make parallel sync database or blocking calls within a single async route

### DISCUSSION THREAD

**Discussion by dhodun** [fastapi/fastapi#6521] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.

### Description

We would like to make multiple synchronous database and API calls in parallel, like we can do in Tornado. I see that you can use `fastapi.concurrency.run_in_threadpool` as the recommended (undocumented) way to run a synchronous call inside an async route. Is there a way to run multiple at the same time, potentially `fastapi.concurrency.iterate_in_threadpool` ?

The code in tornado looked something like this:

```Python
  responses = await multi([
      tornado.ioloop.IOLoop.current().run_in_executor(None, get_document_by_id,
                                                      obj["collection_name"],
                                                      obj["doc_id"])
      for obj in parameters
  ])
```

I asked about this in a closed post as well: https://github.com/tiangolo/fastapi/issues/1066

**Comment by Mause** [comment, marked as answer]
```python
from asyncio import gather

responses = await gather(
    *(
        run_in_threadpool(get_document_by_name, obj["collection_name"], obj["doc_id"])
        for obj in parameters
    )
)
```

the standard library has plenty of nice stuff for this :)

**Comment by dhodun** [comment]
Beautiful, thanks!
